FWCore/Catalog/src/InputFileCatalog.cc

```diff
@@ -54,30 +39,42 @@ namespace edm {
                                         tmps[2],   //desired-data-access-site
                                         tmps[3],   //desired-data-access-volume
                                         tmps[4]);  //desired-data-access-protocol
-
-      overrideFileLocator_ =
-          std::make_unique<FileLocator>(override_struct);  // propagate_const<T> has no reset() function
-    }
-
-    std::vector<CatalogAttributes> const& tmp_dataCatalogs = localconfservice->dataCatalogs();
-    fileLocators_.clear();
-    // Construct all file locators from data catalogs. If a data catalog is
-    // invalid (wrong protocol for example), it is skipped and no file locator
-    // is constructed (an exception is thrown out from FileLocator::init).
-    for (const auto& catalog : tmp_dataCatalogs) {
-      try {
-        fileLocators_.push_back(std::make_unique<FileLocator>(catalog));
-      } catch (cms::Exception const& e) {
-        edm::LogWarning("InputFileCatalog")
-            << "Caught an exception while constructing a file locator in InputFileCatalog::init: " << e.what()
-            << "Skip this catalog";
+      if (override_struct.empty()) {
+        cms::Exception ex("FileCatalog");
```

Comment:
I added this exception. I cannot imagine how someone would intentionally configure an empty override string. This was handled a different way in FileLocator before that made no sense.